ios-base/Navigators/Navigator.swift

```diff
@@ -0,0 +1,204 @@
+   - Parameters:
+   - Navigator: The destination navigator that you want to navigate to
+   - animated: Animate the transition or not.
+   - completion: Completion handler.
+   */
+  func navigate(to router: Navigator, animated: Bool, completion: (() -> Void)?)
+
+  /**
+   Handles backwards navigation through the stack.
+   */
+  func pop(animated: Bool)
+
+  /**
+   Handles backwards navigation through the stack.
+   - Parameters:
+   - route: The index of the route of your navigation action.
+   - animated: Animate the transition or not.
+   */
+  func popTo(index: Int, animated: Bool)
+
+  /**
+   Handles backwards navigation through the stack.
+   - Parameters:
+   - route: The destination route of your navigation action.
+   - animated: Animate the transition or not.
+   */
+  func popTo(route: Route, animated: Bool)
+
+  /**
+   Dismiss your current ViewController.
+   - Parameters:
+   - animated: Animate the transition or not.
+   - completion: Completion handler.
+   */
+  func dismiss(animated: Bool, completion: (() -> Void)?)
+}
+
+public extension Navigator {
+  func navigate(to route: Route, with transition: TransitionType, animated: Bool = true, completion: (() -> Void)? = nil) {
+    let viewController = route.screen
+    switch transition {
+    case .modal:
+      currentViewController?.present(viewController, animated: animated, completion: completion)
+      currentViewController = viewController
+    case .push:
+      rootViewController?.pushViewController(viewController, animated: animated)
+      currentViewController = viewController
+    case .reset:
+      rootViewController?.setViewControllers([viewController], animated: animated)
+      currentViewController = viewController
+    case .changeRoot:
+      UIView.animate(withDuration: 0.3) { [weak self] in
+        if let navigation = (viewController as? UINavigationController) {
+          UIApplication.shared.keyWindow?.rootViewController = navigation
+          self?.rootViewController = navigation
+          self?.currentViewController = self?.rootViewController?.topViewController
+        } else {
+          let navigationController = UINavigationController(rootViewController: viewController)
+          UIApplication.shared.keyWindow?.rootViewController = navigationController
+          self?.rootViewController = navigationController
+          self?.rootViewController?.isNavigationBarHidden = true
+          self?.currentViewController = viewController
+        }
+      }
+    }
+  }
+
+  func navigate(to router: Navigator, animated: Bool, completion: (() -> Void)?) {
+    guard let viewController = router.rootViewController else {
+      assert(false, "Navigator does not have a root view controller")
+      return
+    }
+
+    currentViewController?.present(viewController, animated: animated, completion: completion)
```

Comment:
I think we can use the method above passing the `TransitionType.modal` to avoid repetitions.

Reply:
Actually, we can't since the navigation method receives a `Route`, here, we are using the rootVc of another navigator.

Reply:
So that leaves routing betweeen `Navigator` objects to be modally only?